### PR TITLE
Structure_oM: Adding OrientationAngle to Panel

### LINKED ISSUE #963: Structure_oM: Add OrientationAngle for Panel
#### Description:

The normal from the Panel can be extracted based on its geometry. To control the direction of the local x and y one more parameter is needed though. Should add in an OrientationAngle proeprty, that would work based on similar principles that it does on the Bar to control local x. Local y can then be found via a crossproduct between the normal and local x.

@rwemay @JosefTaylor @peterjamesnugent

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #963 

Adding Orientation angle to Panel and FEMesh.

### Test files

See testfiles for Engine PR

### CODE REVIEW

File: Structure_oM/Elements/FEMeshFace.cs [hunk 1]
```diff
@@ -37,6 +38,15 @@ public class FEMeshFace : BHoMObject, IFace
         [Description("List of which node indices in the host FEMesh node list that this face is connecting.")]
         public virtual List<int> NodeListIndices { get; set; } = new List<int>();
 
+
+        [Angle]
+        [Description("Defines the angle that the local x and y are rotated around the normal, local z, of the FEFace.\n" +
```
Comment (peterjamesnugent) [inline Structure_oM/Elements/FEMeshFace.cs:43]: local x and y axis
around the normal (i.e. local z)

File: Structure_oM/Elements/FEMeshFace.cs [hunk 2]
```diff
@@ -37,6 +38,15 @@ public class FEMeshFace : BHoMObject, IFace
...
+                     "The normal of the FEMeshFace is determined by the right hand rule of the node list.\n" +
+                     "Local x is found by projecting the global X to the plane of the FEMeshFace and is then rotated with the OrientationAngle, " +
+                     "except for the case when the Normal is parallel with the global X. " +
+                     "For this case the local y is instead found by projecting global Y to the plane of the FEMeshFace and is then rotated with the OrientationAngle.\n" +
+                     "The remaining local axis is determained by the right hand rule.")]
```
Comment (peterjamesnugent) [inline Structure_oM/Elements/FEMeshFace.cs:48]: axes 
determined

File: Structure_oM/Elements/Panel.cs [hunk 3]
```diff
@@ -46,8 +47,16 @@ public class Panel : Base.BHoMObject, IAreaElement, IElement2D, IPanel<Edge,Open
         [Description("Defines the thickness property and material of the Panel.")]
         public virtual ISurfaceProperty Property { get; set; } = new ConstantThickness();
 
+        [Angle]
+        [Description("Defines the angle that the local x and y are rotated around the normal, local z, of the Panel.\n" +
+                     "The normal of the Panel is determined by the right hand rule of the edges.\n" +
```
Comment (peterjamesnugent) [inline Structure_oM/Elements/Panel.cs:52]: local x and y axis
around the normal (i.e. local z)

File: Structure_oM/Elements/FEMeshFace.cs [hunk 4]
```diff
@@ -37,6 +38,15 @@ public class FEMeshFace : BHoMObject, IFace
         [Description("List of which node indices in the host FEMesh node list that this face is connecting.")]
         public virtual List<int> NodeListIndices { get; set; } = new List<int>();
 
+
+        [Angle]
+        [Description("Defines the angle that the local x and y are rotated around the normal, local z, of the FEFace.\n" +
+                     "The normal of the FEMeshFace is determined by the right hand rule of the node list.\n" +
```
Comment (peterjamesnugent) [inline Structure_oM/Elements/FEMeshFace.cs:44]: We should clarify this to:
The normal of the FEMeshFace is determined by the curl right hand rule dictated by the order of the node list.

File: Structure_oM/Elements/Panel.cs [hunk 5]
```diff
@@ -46,8 +47,16 @@ public class Panel : Base.BHoMObject, IAreaElement, IElement2D, IPanel<Edge,Open
...
+                     "Local x is found by projecting the global X to the plane of the Panel and is then rotated with the OrientationAngle, " +
+                     "except for the case when the Normal is parallel with the global X. " +
+                     "For this case the local y is instead found by projecting global Y to the plane of the Panel and is then rotated with the OrientationAngle.\n" + 
+                     "The remaining local axis is determained by the right hand rule.")]
```
Comment (peterjamesnugent) [inline Structure_oM/Elements/Panel.cs:56]: axes
determined

File: Structure_oM/Elements/Panel.cs (hunk 3 above)
Comment (peterjamesnugent) [inline Structure_oM/Elements/Panel.cs:52]: The normal of the Panel is determined by the right hand curl rule dictated by the order of the edge list.

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Some small clarifications.

File: Structure_oM/Elements/FEMeshFace.cs (hunk 2 above)
Comment (IsakNaslundBh) [reply Structure_oM/Elements/FEMeshFace.cs]: think it should be axis, as it is the single outstanding?

File: Structure_oM/Elements/Panel.cs (hunk 5 above)
Comment (IsakNaslundBh) [reply Structure_oM/Elements/Panel.cs]: as above

**Comment by IsakNaslundBh** [comment]
Thanks, have ammended!

File: Structure_oM/Elements/FEMeshFace.cs (hunk 2 above)
Comment (peterjamesnugent) [reply Structure_oM/Elements/FEMeshFace.cs]: "The remaining local axis is determined by the right hand rule.")]
- local axes refers to multiple axis (x,y and z) so should be plural.

File: Structure_oM/Elements/Panel.cs (hunk 5 above)
Comment (peterjamesnugent) [reply Structure_oM/Elements/Panel.cs]: "The remaining local axis is determined by the right hand rule.")]
- local axes refers to multiple axis (x,y and z) so should be plural.

**Review by peterjamesnugent — APPROVED** [review]
LGTM.

File: Structure_oM/Elements/FEMeshFace.cs [hunk 6]
```diff
@@ -37,6 +38,15 @@ public class FEMeshFace : BHoMObject, IFace
         [Description("List of which node indices in the host FEMesh node list that this face is connecting.")]
         public virtual List<int> NodeListIndices { get; set; } = new List<int>();
 
+
+        [Angle]
+        [Description("Defines the angle that the local x and y axes are rotated around the normal (i.e. local z) of the FEFace.\n" +
```
Comment (pawelbaran) [inline Structure_oM/Elements/FEMeshFace.cs:43]: FEFace sounds a bit weird (and is inconsistent with further lines)

File: Structure_oM/Elements/FEMeshFace.cs [hunk 7]
```diff
@@ -37,6 +38,15 @@ public class FEMeshFace : BHoMObject, IFace
         [Description("List of which node indices in the host FEMesh node list that this face is connecting.")]
         public virtual List<int> NodeListIndices { get; set; } = new List<int>();
 
+
```
Comment (pawelbaran) [inline Structure_oM/Elements/FEMeshFace.cs:41]: one blank too much

File: Structure_oM/Elements/FEMeshFace.cs [hunk 8]
```diff
@@ -37,6 +38,15 @@ public class FEMeshFace : BHoMObject, IFace
...
+        [Angle]
+        [Description("Defines the angle that the local x and y axes are rotated around the normal (i.e. local z) of the FEFace.\n" +
+                     "The normal of the FEMeshFace is determined by the curl right hand rule dictated by the order of the node list.\n" +
+                     "Local x is found by projecting the global X to the plane of the FEMeshFace and is then rotated with the OrientationAngle, " +
+                     "except for the case when the Normal is parallel with the global X. " +
+                     "For this case the local y is instead found by projecting global Y to the plane of the FEMeshFace and is then rotated with the OrientationAngle.\n" +
+                     "The remaining local axis is determined by the right hand rule.")]
+        public virtual double OrientationAngle { get; set; } = 0;
```
Comment (pawelbaran) [inline Structure_oM/Elements/FEMeshFace.cs:49]: missing blank line

File: Structure_oM/Elements/Panel.cs [hunk 9]
```diff
@@ -46,8 +47,16 @@ public class Panel : Base.BHoMObject, IAreaElement, IElement2D, IPanel<Edge,Open
         [Description("Defines the thickness property and material of the Panel.")]
```
Comment (pawelbaran) [inline Structure_oM/Elements/Panel.cs:47]: a word about the local CS could be added

File: Structure_oM/Elements/FEMeshFace.cs [hunk 10]
```diff
@@ -24,6 +24,7 @@
 using BH.oM.Analytical.Elements;
 using BH.oM.Base;
 using System.ComponentModel;
+using BH.oM.Quantities.Attributes;
 
 namespace BH.oM.Structure.Elements
 {
```
Comment (pawelbaran) [inline Structure_oM/Elements/FEMeshFace.cs:30]: _All geometrical information is stored on the host FEMesh._ - not sure if that part of the description is still true

**Review by pawelbaran — COMMENTED** [review]
LGTM in general, a few improvement proposed in the code comments (therefore no block, but no approval 😉).

**Review by pawelbaran — APPROVED** [review]
LGTM now 👍

**Pull request merged**